src/ray/rpc/worker/core_worker_client.cc

```diff
@@ -0,0 +1,134 @@
+      ::RayConfig::instance().grpc_client_check_connection_status_interval_milliseconds(),
+      /*server_unavailable_timeout_seconds=*/
+      ::RayConfig::instance().core_worker_rpc_server_reconnect_timeout_s(),
+      /*server_unavailable_timeout_callback=*/
+      core_worker_unavailable_timeout_callback,
+      /*server_name=*/"Core worker " + addr_.ip_address());
+}
+
+void CoreWorkerClient::PushActorTask(std::unique_ptr<PushTaskRequest> request,
+                                     bool skip_queue,
+                                     ClientCallback<PushTaskReply> &&callback) {
+  if (skip_queue) {
+    // Set this value so that the actor does not skip any tasks when
+    // processing this request. We could also set it to max_finished_seq_no_,
+    // but we just set it to the default of -1 to avoid taking the lock.
+    request->set_client_processed_up_to(-1);
+    INVOKE_RPC_CALL(CoreWorkerService,
+                    PushTask,
+                    *request,
+                    callback,
+                    grpc_client_,
+                    /*method_timeout_ms*/ -1);
+    return;
+  }
+
+  {
+    absl::MutexLock lock(&mutex_);
+    send_queue_.emplace_back(std::move(request), std::move(callback));
+  }
+  SendRequests();
+}
+
+void CoreWorkerClient::PushNormalTask(std::unique_ptr<PushTaskRequest> request,
+                                      const ClientCallback<PushTaskReply> &callback) {
+  request->set_sequence_number(-1);
+  request->set_client_processed_up_to(-1);
+  INVOKE_RPC_CALL(CoreWorkerService,
+                  PushTask,
+                  *request,
+                  callback,
+                  grpc_client_,
+                  /*method_timeout_ms*/ -1);
+}
+
+/// Get the estimated size in bytes of the given task.
+static const int64_t RequestSizeInBytes(const PushTaskRequest &request) {
+  int64_t size = kBaseRequestSize;
+  for (auto &arg : request.task_spec().args()) {
+    size += arg.data().size();
+  }
+  return size;
+}
+
+void CoreWorkerClient::SendRequests() {
+  absl::MutexLock lock(&mutex_);
+  auto this_ptr = this->shared_from_this();
+
+  while (!send_queue_.empty() && rpc_bytes_in_flight_ < kMaxBytesInFlight) {
+    auto pair = std::move(*send_queue_.begin());
+    send_queue_.pop_front();
+
+    auto request = std::move(pair.first);
+    int64_t task_size = RequestSizeInBytes(*request);
+    int64_t seq_no = request->sequence_number();
+    request->set_client_processed_up_to(max_finished_seq_no_);
+    rpc_bytes_in_flight_ += task_size;
+
+    auto rpc_callback =
+        [this, this_ptr, seq_no, task_size, callback = std::move(pair.second)](
+            Status status, rpc::PushTaskReply &&reply) {
+          {
+            absl::MutexLock lock(&mutex_);
+            if (seq_no > max_finished_seq_no_) {
+              max_finished_seq_no_ = seq_no;
+            }
+            rpc_bytes_in_flight_ -= task_size;
+            RAY_CHECK(rpc_bytes_in_flight_ >= 0);
```

Comment:
`RAY_CHECK_GE`

Reply:
Use `RAY_CHECK_GE` so that the value of `rpc_bytes_in_flight_` will be printed.